### PR TITLE
Add sdl2_image library

### PR DESCRIPTION
**Pull request by patrigg**
SDL2_image adds extended image loading capabilities to SDL2 allowing many more file formats to be loaded into an SDL_Surface, like PNG, JPEG, TIFF, WEBP,...

This port enables all file formats which SDL2_image can handle internally. PNG, JPEG, TIFF and WEBP file formats are supported through external libraries. This port will automatically link to all installed libraries at the time SDL2_image is installed.
If a library is not available the port will fall back to linking at run time. In this case, dlls have to be copied manually to the application directory since vcpkg cannot resolve them.

I used the FindWEBP.cmake file from the third party WebKit library for resolving the dependency to WebP.
Submission containing materials of a third party:
FindWEBP.cmake by Raphael Kubo da Costa <rakuco@webkit.org> and Igalia S.L. taken from WebKit: https://github.com/WebKit/webkit/blob/master/Source/cmake/FindWebP.cmake

### CODE REVIEW

File: ports/sdl2-image/FindWEBP.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+find_path(WEBP_INCLUDE_DIRS
+    NAMES webp/decode.h
+    HINTS ${PC_WEBP_INCLUDEDIR} ${PC_WEBP_INCLUDE_DIRS}
+)
+mark_as_advanced(WEBP_INCLUDE_DIRS)
+
+# Look for the library.
+find_library(
+    WEBP_LIBRARIES
+    NAMES webp
```
Comment (codicodi) [inline ports/sdl2-image/FindWEBP.cmake]: Will this find debug version? It is named `webpd`.

Comment (patrigg) [inline ports/sdl2-image/FindWEBP.cmake]: Thanks for reviewing. No, it won't find the debug version.

**Comment by patrigg** [comment]
The issue with finding the debug version is now fixed.

**Comment by traversaro** [comment]
For reference, I wrote a [`FindWEBP.cmake` file](https://github.com/Microsoft/vcpkg/blob/master/ports/freeimage/cmake/FindWEBP.cmake) while writing the `freeimage` port. I don't think it is a big problem to have two different `FindWEBP.cmake` files in different ports, I wanted just to point out to the already existing one (that has no copyright attribution problems, as I wrote it ex-novo).

**Comment by traversaro** [comment]
> This port enables all file formats which SDL2_image can handle internally. PNG, JPEG, TIFF and WEBP file formats are supported through external libraries. This port will automatically link to all installed libraries at the time SDL2_image is installed.
If a library is not available the port will fall back to linking at run time. In this case, dlls have to be copied manually to the application directory since vcpkg cannot resolve them.

I don't think it is a good idea to have a port that change what is building depending on the state of the vcpkg installation, as it would be extremely difficult to debug all possible combination of available/not available ports (see for example https://github.com/Microsoft/vcpkg/issues/412 ).
I think it is cleaner to just add `libpng, libjpeg-turbo, tiff, libwebp` to `Build-Depends:` to be sure that all dependencies are always available, and always link them in the "classical" way.

**Comment by patrigg** [comment]
I switched to your version of FindWEB.cmake. However, why are you setting WEBP_LIBRARIES to WEBP_LIBRARY? Both variables should be set correctly by select_library_configurations.

Though I think the current approach with linking is more flexible, I get your point regarding issues with bugs based on port availability. I added the build dependencies as you suggested.

**Comment by traversaro** [comment]
I think you are right about setting WEB_LIBRARY and setting WEBP_LIBRARIES to WEBP_LIBRARY , those two lines can be probably be removed.

**Comment by codicodi** [comment]
I'd also like to point out that installation dir for the header file seems quite nonstandard - most of the other distributions put it in include/SDL2.

**Comment by patrigg** [comment]
@codicodi changed the installation dir as you suggested

**Comment by patrigg** [comment]
I squashed my commits. Is there anything I need to do to?

**Comment by ras0219-msft** [comment]
Awesome library, thanks for the port. GitHub was not diffing correctly so it took a bit longer to review :).

**Pull request merged**